x/gov/keeper/proposal_test.go

```diff
@@ -160,6 +160,52 @@ func (suite *KeeperTestSuite) TestGetProposalsFiltered() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestCancelProposal() {
+	govAcct := suite.govKeeper.GetGovernanceAccount(suite.ctx).GetAddress().String()
+	tp := v1beta1.TextProposal{Title: "title", Description: "description"}
+	prop, err := v1.NewLegacyContent(&tp, govAcct)
+	suite.Require().NoError(err)
+	proposalResp, err := suite.govKeeper.SubmitProposal(suite.ctx, suite.addrs[0], []sdk.Msg{prop}, "")
+	proposalID := proposalResp.Id
+
+	testCases := []struct {
+		name        string
+		proposalID  uint64
+		proposer    string
+		expectedErr bool
+	}{
+		{
+			name:        "invalid proposal id",
+			proposalID:  1,
+			proposer:    string(suite.addrs[0]),
+			expectedErr: true,
+		},
+		{
+			name:        "valid proposalID but invalid proposer",
+			proposalID:  proposalID,
+			proposer:    suite.addrs[1].String(),
+			expectedErr: true,
+		},
+		{
+			name:        "valid proposer and proposal id",
+			proposalID:  proposalID,
+			proposer:    suite.addrs[0].String(),
+			expectedErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			err = suite.govKeeper.CancelProposal(suite.ctx, tc.proposalID, tc.proposer)
```

Comment:
How about calling the msg_server's CancelProposal function instead of the keeper's one here?

This way we can test canceledTime and canceledHeight too